reproducibility_tests/latest_comparable_paths.jl

```diff
@@ -0,0 +1,110 @@
+import Dates
+
+function sorted_dataset_folder(; dir = pwd())
+    matching_paths = String[]
+    for file in readdir(dir)
+        !ispath(joinpath(dir, file)) && continue
+        push!(matching_paths, joinpath(dir, file))
+    end
```

Comment:
```suggestion
    matching_paths = filter(isdir, readdir(dir; join=true))
```
I think that this is equivalent, if I understand the intent of `!ispath`

Reply:
I think `matching_paths = filter(ispath, readdir(dir; join=true))` is equivalent, `isdir` returns `false` for files:

```julia
help?> isdir
search: isdir isdirpath isdigit isdisjoint isdispatchtuple isxdigit isdefined @isdefined redirect_stdio redirect_stdin

  isdir(path) -> Bool

  Return true if path is a directory, false otherwise.

  Examples
  ≡≡≡≡≡≡≡≡

  julia> isdir(homedir())
  true
  
  julia> isdir("not/a/directory")
  false

  See also isfile and ispath.
```